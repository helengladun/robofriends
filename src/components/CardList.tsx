import React from 'react';

// components
import Card from "./Card";

// interfaces
import {IRobot} from '../interfaces/IRobot'

interface IProps {
  robots: IRobot[]
}

const CardList = ({robots}: IProps) => (
  <div>
    {robots.map(robot => <Card key={robot.id} data={robot} />)}
  </div>
);

export default CardList;
