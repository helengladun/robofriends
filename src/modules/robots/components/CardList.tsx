import React from 'react';

// components
import Card from "./Card";

// interfaces
import {IRobot} from '../../shared/models/IRobot'

interface IProps {
  robots: IRobot[]
}

const CardList = ({robots}: IProps) => (
  <div>
    {robots.map(robot => <Card key={robot.id} data={robot} />)}
  </div>
);

export default CardList;
