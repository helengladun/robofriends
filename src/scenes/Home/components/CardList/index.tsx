import React from 'react';

// components
import Card from "../Card/index";

// interfaces
import {IRobot} from '../../../models/IRobot';

interface IProps {
  robots: IRobot[]
}

const CardList = ({robots}: IProps) => (
  <div>
    {robots.map(robot => <Card key={robot.id} data={robot} />)}
  </div>
);

export default React.memo(CardList);
CardList.whyDidYouRender = true;
