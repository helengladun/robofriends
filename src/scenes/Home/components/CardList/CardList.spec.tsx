import React from 'react';
import { shallow } from 'enzyme';
import CardList from './index';

describe('CardList component', () => {
  const listMock = [
    {
      id: 1,
      name: 'user1',
      email: 'user1@gmail.com'
    },
    {
      id: 2,
      name: 'user2',
      email: 'user2@gmail.com'
    }
  ];

  it('render CardList component correctly', () => {
    expect(shallow(<CardList robots={listMock} />)).toMatchSnapshot();
  });
});