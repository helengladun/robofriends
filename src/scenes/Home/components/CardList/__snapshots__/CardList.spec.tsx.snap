// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component render CardList component correctly 1`] = `
<div>
  <Memo(Card)
    data={
      Object {
        "email": "user1@gmail.com",
        "id": 1,
        "name": "user1",
      }
    }
    key="1"
  />
  <Memo(Card)
    data={
      Object {
        "email": "user2@gmail.com",
        "id": 2,
        "name": "user2",
      }
    }
    key="2"
  />
</div>
`;
